ethcore/src/state/mod.rs

```diff
@@ -108,6 +108,20 @@ impl AccountEntry {
 			state: AccountState::Clean,
 		}
 	}
+
+	// Replace data with another entry but preserve storage cache
+	fn merge_snapshot(&mut self, other: AccountEntry) {
+		self.state = other.state;
```

Comment:
just to clarify, the purpose is basically to completely replace the cached account with the snapshotted block, while also preserving any new cached storage values since the snapshotted version was put away?